tests/system/providers/amazon/aws/tests/test_aws_auth_manager.py

```diff
@@ -0,0 +1,211 @@
+            ("aws_auth_manager", "region_name"): region_name,
+            ("aws_auth_manager", "saml_metadata_url"): SAML_METADATA_URL,
+            ("aws_auth_manager", "avp_policy_store_id"): avp_policy_store_id,
+        }
+    ):
+        with patch(
+            "airflow.providers.amazon.aws.auth_manager.views.auth.OneLogin_Saml2_IdPMetadataParser"
+        ) as mock_parser, patch(
+            "airflow.providers.amazon.aws.auth_manager.views.auth.AwsAuthManagerAuthenticationViews._init_saml_auth"
+        ) as mock_init_saml_auth:
+            mock_parser.parse_remote.return_value = SAML_METADATA_PARSED
+
+            yield mock_init_saml_auth
+
+
+@pytest.fixture
+def client_no_permissions(base_app):
+    auth = Mock()
+    auth.is_authenticated.return_value = True
+    auth.get_nameid.return_value = "user_no_permissions"
+    auth.get_attributes.return_value = {
+        "id": ["user_no_permissions"],
+        "groups": [],
+        "email": ["email"],
+    }
+    base_app.return_value = auth
+    return application.create_app(testing=True)
+
+
+@pytest.fixture
+def client_admin_permissions(base_app):
+    auth = Mock()
+    auth.is_authenticated.return_value = True
+    auth.get_nameid.return_value = "user_admin_permissions"
+    auth.get_attributes.return_value = {
+        "id": ["user_admin_permissions"],
+        "groups": ["Admin"],
+    }
+    base_app.return_value = auth
+    return application.create_app(testing=True)
+
+
+@pytest.mark.system("amazon")
+class TestAwsAuthManager:
+    """
+    Run tests on Airflow using AWS auth manager with real credentials
+    """
+
+    @classmethod
+    def teardown_class(cls):
+        cls.delete_avp_policy_store()
+
+    @classmethod
+    def delete_avp_policy_store(cls):
+        client = boto3.client("verifiedpermissions")
+
+        paginator = client.get_paginator("list_policy_stores")
+        pages = paginator.paginate()
+        policy_store_ids = [
+            store["policyStoreId"]
+            for page in pages
+            for store in page["policyStores"]
+            if "description" in store
+            and f"Created by system test TestAwsAuthManager: {env_id_cache}" in store["description"]
+        ]
+
+        for policy_store_id in policy_store_ids:
+            client.delete_policy_store(policyStoreId=policy_store_id)
+
+    def test_login_no_permissions(self, client_no_permissions):
+        with client_no_permissions.test_client() as client:
+            response = client.get("/login_callback", follow_redirects=True)
+            check_content_in_response("Your user has no roles and/or permissions!", response, 403)
+
+    def test_login_admin(self, client_admin_permissions):
+        with client_admin_permissions.test_client() as client:
+            response = client.get("/login_callback", follow_redirects=True)
+            print(response.data.decode("utf-8"))
+            check_content_in_response("<h2>DAGs</h2>", response, 200)
```

Comment:
Why should this be in the response after the login? Is this detecting that you land on the airflow home page with the DAGs listed?

Reply:
Correct :)